siliconcompiler/core.py

```diff
@@ -3015,6 +3032,14 @@ def node(self, flow, step, tool, index=0):
 
         # bind tool to node
         self.set('flowgraph', flow, step, str(index), 'tool', tool)
+
+        tasks = self.getkeys('tool', tool, 'task')
+
+        if task is None:
+            task = step
```

Comment:
Not a huge fan of blindly setting this, if task is a requirement, then this is not needed. I think you could go the other way since step is just a name and can be anything, so 
```
if step is None:
  step = task
```